tests/unit/service/test_observation_table.py

```diff
@@ -128,48 +138,80 @@ async def mock_list_table_schema(*args, **kwargs):
     )
     with pytest.raises(MissingPointInTimeColumnError):
         await observation_table_service.validate_materialized_table_and_get_metadata(
-            mock_db_session, table_details
+            mock_db_session, table_details, snowflake_feature_store
         )
 
 
 @pytest.mark.asyncio
 async def test_validate__most_recent_point_in_time(
-    observation_table_service, db_session, table_details, cust_id_entity
+    observation_table_service,
+    db_session,
+    table_details,
+    cust_id_entity,
+    snowflake_feature_store,
+    insert_credential,
 ):
     """
     Test validate_materialized_table_and_get_metadata triggers expected query
     """
-    _ = cust_id_entity
-    metadata = await observation_table_service.validate_materialized_table_and_get_metadata(
-        db_session, table_details
-    )
+    _ = cust_id_entity, insert_credential
+    with mock.patch(
+        "featurebyte.service.preview.PreviewService._get_feature_store_session"
+    ) as mock_get_feature_store_session:
+        mock_get_feature_store_session.return_value = (snowflake_feature_store, db_session)
+        metadata = await observation_table_service.validate_materialized_table_and_get_metadata(
+            db_session, table_details, snowflake_feature_store
+        )
 
-    expected_query = textwrap.dedent(
-        """
-        SELECT
-          MAX("POINT_IN_TIME")
-        FROM "fb_database"."fb_schema"."fb_table"
-        """
-    ).strip()
-    query = db_session.execute_query.call_args[0][0]
-    assert query == expected_query
+        expected_query = textwrap.dedent(
```

Comment:
the query here has changed to be the last query executed (which is the one to get the describe stats)

Reply:
Would it make sense to check all the relevant queries ran if there are more than one?

Reply:
i couldn't find it in the `call_args` param, let me take a closer look tomorrow.

Reply:
There is a `call_args_list` attribute that records all the calls made.